src/elmAnalyse.ts

```diff
@@ -0,0 +1,240 @@
+import * as vscode from 'vscode';
+import * as cp from 'child_process';
+import WebSocket = require("ws");
+import * as utils from './elmUtils';
+import * as path from 'path';
+import * as readline from 'readline';
+import {isWindows, execCmd, ExecutingCmd } from './elmUtils';
+import {runLinter, IElmIssue} from './elmLinter';
+
+export class ElmAnalyse {
+  public constructor(public elmAnalyseIssues: IElmIssue[]) {
+    this.statusBarStopButton = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Left);
+    this.statusBarInformation = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Left);
+    this.statusBarStopButton.text = '$(primitive-square)' + ' Stop Elm-analyse process';
+    this.statusBarStopButton.command = 'elm.analyseStop';
+    this.statusBarStopButton.tooltip = 'Stop elm-analyse process';
+    this.analyse = {} as ExecutingCmd;
+    this.messageDescriptionsMap = new Map()
+    this.messageDescriptions.forEach(element => {
+      this.messageDescriptionsMap.set(element.messageType, element.description)
+    });
+    const config: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration('elm');
+    const enabledOnStartup: boolean = <boolean>config.get('analyseEnabled');
+    if (enabledOnStartup) {
+      this.activateAnalyseProcesses();
+    }
+   }
+  private statusBarStopButton: vscode.StatusBarItem;
+  private statusBarInformation: vscode.StatusBarItem;
+  private analyseSocket: WebSocket;
+  private analyse: ExecutingCmd;
+  private oc: vscode.OutputChannel = vscode.window.createOutputChannel('Elm Analyse');
+  private messageDescriptionsMap : Map<string, string>
```

Comment:
I'm not that familiar with Type Script, but shouldn't this line be closed with a semicolon?